### PR TITLE
refactor: controller details header

### PR DESCRIPTION
**Pull request by petermakowski**
## Done

- build controller details header

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the Controller details page
- Click on the Controller's name and verify the hostname cannot be changed
- Change the domain using the dropdown and verify it's been saved correctly

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/932

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

Kapture 2022-06-02 at 20 21 09

### CODE REVIEW

File: ui/src/app/base/components/NodeName/_index.scss [hunk 1]
```diff
@@ -45,6 +45,16 @@
     white-space: pre;
   }
 
+  .node-name__hostname--no-edit {
+    @extend %vf-heading-4;
+
+    @media only screen and (min-width: $breakpoint-large) {
+      padding: $sp-xx-small 0 calc(#{$sp-xx-small} + 1px) calc(#{$spv--large} + 1px);
```
Comment (Caleb-Ellis) [inline ui/src/app/base/components/NodeName/_index.scss:52]: Down to the pixel! Wild

**Review by Caleb-Ellis — APPROVED** [review]
LGTM!

File: ui/src/app/base/components/NodeName/NodeName.tsx [hunk 2]
```diff
@@ -10,15 +10,17 @@ import NodeNameFields from "./NodeNameFields";
...
 import type { Domain, DomainMeta } from "app/store/domain/types";
 import type { Machine } from "app/store/machine/types";
 import type { SimpleNode } from "app/store/types/node";
+import { nodeIsController } from "app/store/utils";
 
 export type Props = {
   editingName: boolean;
   // Machines and devices can edit their name, but no controllers.
-  node: Machine | Device | null;
+  node: Machine | Device | Controller | null;
```
Comment (huwshimi) [inline ui/src/app/base/components/NodeName/NodeName.tsx:23]: We also have a `Node` type that covers these three:

https://github.com/canonical-web-and-design/maas-ui/blob/main/ui/src/app/store/types/node.ts#L209

Comment (petermakowski) [reply ui/src/app/base/components/NodeName/NodeName.tsx:23]: Ah yeah good point 😄

**Pull request merged**